cdap-ui/app/cdap/components/DataPrep/Directives/Parse/Modals/SingleFieldModal.js

```diff
@@ -30,9 +30,28 @@ export default class SingleFieldModal extends Component {
     };
 
     this.onTextChange = this.onTextChange.bind(this);
+    this.onOptionalTextChange = this.onOptionalTextChange.bind(this);
     this.apply = this.apply.bind(this);
   }
 
+  componentDidMount() {
+    this.allMouseTrapInstances = [];
+    document
+      .querySelectorAll('.modal .modal-content .form-control')
+      .forEach(ele => this.allMouseTrapInstances.push(new MouseTrap(ele)));
+    this.allMouseTrapInstances
+      .forEach(mtInstance => mtInstance.bind('enter', this.apply));
+    MouseTrap.bind('enter', this.apply);
+    this.inputRef.focus();
+  }
+
+  componentWillUnmount() {
+    MouseTrap.reset();
+    if (Array.isArray(this.allMouseTrapInstances) && this.allMouseTrapInstances.length) {
+      this.allMouseTrapInstances.forEach(mtInstance => mtInstance.reset());
```

Comment:
Why did you use `reset()` here and a few lines above instead of `unbind('enter')`?